code/modules/mob/living/carbon/human/death.dm

```diff
@@ -43,13 +44,13 @@
 		var/datum/effect_system/spark_spread/s = new /datum/effect_system/spark_spread
 		s.set_up(3, 1, src)
 		s.start()
-
-	spawn(15)
-		if(animation)	qdel(animation)
-		if(src)			qdel(src)
+	QDEL_IN(animation, 15)
+	QDEL_IN(src, 0)

```

Comment:
you could just make this `qdel(src)`. I'm pretty sure `qdel` returns if whatever you pass it is `null`.

Reply:
[If the thing is not a datum, it'll get passed directly to `del`](https://github.com/ParadiseSS13/Paradise/blob/master/code/controllers/subsystem/garbage.dm#L289)

I recall seeing runtime logs of `del` being called with `null` - so I don't think it's safe to pass `null` to qdel

However, I used `QDEL_IN` not to bypass the case of `src` being `null`, but rather to bypass the `spawn` in use here.